src/tiered/tiered_handle.c

```diff
@@ -361,10 +377,37 @@ __tiered_switch(WT_SESSION_IMPL *session, const char *config)
     tracking = false;
     WT_ERR(__wt_meta_track_off(session, true, ret != 0));
     WT_ERR(__tiered_update_dhandles(session, tiered));
+
+    if (tiered->current_id > 1 && tiered->current_id != prev_id) {
+        /*
+         * This code should be done in its own thread, as it may incur network delays or failure.
+         */
+        WT_ERR(__wt_tiered_name(session, &tiered->iface, prev_id, WT_TIERED_NAME_LOCAL, &prev_uri));
+        WT_ERR(__wt_tiered_name(
+          session, &tiered->iface, prev_id, WT_TIERED_NAME_OBJECT, &new_object_uri));
+
+        storage_source = tiered->bstorage->storage_source;
+        bucket_fs = tiered->bstorage->file_system;
+
+        prev_filename = prev_uri;
+        WT_PREFIX_SKIP_REQUIRED(session, prev_filename, "file:");
+        new_object_name = new_object_uri;
+        WT_PREFIX_SKIP_REQUIRED(session, new_object_name, "object:");
```

Comment:
I think we may want a new no-prefix name option to `wt_tiered_name`, maybe `WT_TIERED_NAME_ONLY` that doesn't prepend the prefix. With the changes I merged today for the new suffix, `prev_filename` and `new_object_name` should be identical always. The same object names and the prefix indicates where it lives. Does that mean `ss_flush*` should only have one name arg instead of two?